components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -849,11 +854,66 @@ public static String getUserConsentURL(OAuth2Parameters params, String loggedInU
             }
         } catch (UnsupportedEncodingException e) {
             throw new OAuthSystemException("Error while encoding the url", e);
+        } catch (IdentityOAuth2Exception e) {
+            throw new OAuthSystemException("Error retrieve Service Provider for clientId:" + params.getClientId() , e);
         }
 
         return consentPageUrl;
     }
 
+
+    private static ServiceProvider getServiceProvider(OAuth2Parameters params) throws IdentityOAuth2Exception {
+
+        ServiceProvider sp = null;
+        if (params != null) {
+            sp = OAuth2Util.getServiceProvider(params.getClientId());
+        }
+        return sp;
+    }
+
+    private static boolean isExternalizedConsentPageEnabledForSP(ServiceProvider sp) {
+
+        if (sp == null) {
+            throw new IllegalArgumentException("A null reference received for service provider.");
+        }
+        boolean isEnabled = false;
+        LocalAndOutboundAuthenticationConfig config = sp.getLocalAndOutBoundAuthenticationConfig();
+        if (config != null && config.getExternalizedConsentPageConfig() != null) {
+            isEnabled = config.getExternalizedConsentPageConfig().isEnabled();
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("externalConsentManagement: " + isEnabled + " for application: " +
+                    sp.getApplicationName() + " with id: " + sp.getApplicationID());
+        }
+
+        return isEnabled;
+    }
+
+    private static String getExternalConsentUrlForSP(ServiceProvider sp) throws OAuthSystemException {
+
+        if (sp == null) {
+            throw new IllegalArgumentException("A null reference received for service provider.");
```

Comment:
A better error message would be good.

Reply:
Resolved by https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2030/commits/622be8b0e8069d5383ccb968ef7703a90d2c9d44